scalalib/src/mill/scalalib/GenIdeaImpl.scala

```diff
@@ -225,8 +226,9 @@ object GenIdeaImpl {
       Tuple2(
         ".idea"/"modules.xml",
         allModulesXmlTemplate(
-          for((path, mod) <- modules)
-            yield moduleName(path)
+          modules
+            .filter(!_._2.skipIdea)
```

Comment:
This could use a collect instead to avoid iterating over the modules twice
```
.collect {
  case (path, mod) if !mod.skipIdea => moduleName(path)
}
```